src/hooks/useHasuraQuery.tsx

```diff
@@ -0,0 +1,11 @@
+import { QueryHookOptions, TypedDocumentNode, useQuery } from "@apollo/client";
+import merge from "lodash/merge";
+import { USER_ROLES } from "src/types";
+
+export const useHasuraQuery = (
```

Comment:
Maybe take the whole arguments as an object so that we can provide any part of it independently from the others? Here if I want to provide `options` I also have to provide a `role`

Reply:
Yes good point !